src/surface/Drawing.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable no-empty-function */
+/* eslint-disable no-useless-constructor */
+import { Style } from '../launchpads/base/ILaunchpad.js';
+import { allXy } from '../utils.js';
+import { ILayer } from './ILayer.js';
+
+/**
+ * Helper routines for drawing on a layer
+ */
+export class Drawing {
+  constructor(private readonly layer: ILayer) {
+  }
+
+  /**
+   * Draw a bitmap on the layer at the given coordinates
+   *
+   * The bitmap is specified as a list of strings. Characters in
+   * the list of strings can be mapped to button styles.
+   */
+  public bitmap(xy: [number, number], lines: string[], styleMap: Record<string, Style>): void {
```

Comment:
What does a "bitmap" do exactly?

Reply:
It's to evoke a comparison to bitmap blitting on graphics systems. It allows drawing complex shapes in a minimalist and readable way. An example here:

```js
  new Drawing(surface).bitmap([1, 3], [
    ' x   - ',
    '   o   ',
    'x     x',
    ' xxxxx ',
  ], {
    'x': { style: 'palette', color: 13 }, // yellow
    'o': { style: 'palette', color: 50 }, // purple
    '-': { style: 'pulse', color: 13 }, // purple
  });
```

You can see the shape of the smily-face in the code.

An alternative name for this method could be [`sprite()`](https://en.wikipedia.org/wiki/Sprite_(computer_graphics)) if you don't care for the word `bitmap`.

Reply:
Ah I see